python/cudf_polars/cudf_polars/utils/versions.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Version utilities so that cudf_polars supports a range of polars versions."""
+
+# ruff: noqa: SIM300
+from __future__ import annotations
+
+from packaging.version import parse
+
+from polars import __version__
+
+POLARS_VERSION = parse(__version__)
+
+POLARS_VERSION_GE_10 = POLARS_VERSION >= parse("1.0")
+POLARS_VERSION_GE_11 = POLARS_VERSION >= parse("1.1")
+POLARS_VERSION_GE_12 = POLARS_VERSION >= parse("1.2")
+POLARS_VERSION_GT_10 = POLARS_VERSION > parse("1.0")
+POLARS_VERSION_GT_11 = POLARS_VERSION > parse("1.1")
+POLARS_VERSION_GT_12 = POLARS_VERSION > parse("1.2")
+
+POLARS_VERSION_LE_12 = POLARS_VERSION <= parse("1.2")
+POLARS_VERSION_LE_11 = POLARS_VERSION <= parse("1.1")
+POLARS_VERSION_LT_12 = POLARS_VERSION < parse("1.2")
+POLARS_VERSION_LT_11 = POLARS_VERSION < parse("1.1")
+
+if POLARS_VERSION < parse("1.0"):  # pragma: no cover
```

Comment:
This is not possible, I think, because our pyproject.toml will enforce a minimum version.

Reply:
My long experience suggests that people will litter `--no-deps` everywhere to get to a state where things are installed, but not working.

Reply:
Hm, then we should probably add something for cudf too (make sure users aren't installing an older cudf).

If you want to test this, we can do some monkeypatching where we set the version of polars to something different.

Reply:
> Hm, then we should probably add something for cudf too (make sure users aren't installing an older cudf).

I don't care so much about this. I'm happy to remove this bit as "unreachable code" in normal circumstances, but it's sort of "belt-and-braces".